### PR TITLE
feat(snap): add go-based configure and install hooks

### PR DESCRIPTION
**Pull request by siggiskulason**
This commit adds go-based configure and install hooks to the
snap which adds additional support for environment variable
configuration overrides (see snap/README.md for details).

This commit also adds a content interface plug to allow
configuration.toml, devices, and profiles to be provided
by an external snap.

Signed-off-by: Siggi Skulason <siggi.skulason@canonical.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rest-go/blob/master/.github/CONTRIBUTING.md

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## New Imports

- [ ] Yes
- [ ] No

### CODE REVIEW

File: snap/local/hooks/cmd/install/install.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	err = installDevProfiles()
+	if err != nil {
+		hooks.Error(fmt.Sprintf("edgex-device-rest:install: %v", err))
+		os.Exit(1)
+	}
+
+	// disable the service and handle the autostart logic in the configure hook
+	// as default snap configuration is not available when the install hook runs
+	// TODO: update the service name to drop the "-go"
+	err = cli.Stop("device-rest", true)
```
Comment (tonyespy) [inline snap/local/hooks/cmd/install/install.go:120]: Same as the other two PRs re: disabling service by default...

**Review by tonyespy — CHANGES_REQUESTED** [review]
Same issue as device-modbus wrt service disable in `snapcraft.yaml`.

**Review by tonyespy — APPROVED** [review]
LGTM

**Review by tonyespy — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]

**Pull request merged**